uktrade/data-hub-frontend
# Refactor the ManageAdviser component with TaskForm

## Description of change

Refactors the _add lead ITA_ form with the `TaskForm` component.

## Test instructions

1. Go to `/companies/{id}/advisers/assign`
2. Tweak the [`postAddDnbCompany `](https://github.com/uktrade/data-hub-frontend/blob/master/src/apps/companies/apps/add-company/controllers.js#L68-L83) request handler of the form's POST request to throw an error
2. Try out all the possible user journeys of the form
3. They should all work as before
4. When you submit the form, the task should reject
5. You should see an error message overlaid on top of the form

## Screenshots
### Before
No error handling for failed request:
<img width="584" alt="Screen Shot 2021-11-05 at 2 34 48 PM" src="https://user-images.githubusercontent.com/2333157/140528178-1e03ca9a-3349-4f2a-b6b4-fcb16d2a98ed.png">

### After
<img width="589" alt="Screen Shot 2021-11-05 at 2 39 13 PM" src="https://user-images.githubusercontent.com/2333157/140528308-8a5ff9d3-02ca-417b-9246-0eec78f08ec0.png">

_Add a screenshot_
